File: DuDoong-Domain/src/main/java/band/gosrock/domain/common/events/event/EventCreationEvent.java
package band.gosrock.domain.common.events.event;


import band.gosrock.domain.common.aop.domainEvent.DomainEvent;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class EventCreationEvent extends DomainEvent {
    private final Long hostId;
    private final String eventName;

    public static EventCreationEvent of(Long hostId, String eventName) {
        return EventCreationEvent.builder().hostId(hostId).eventName(eventName).build();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/common/events/event/EventCreationEvent.kt
package band.gosrock.domain.common.events.event

import band.gosrock.domain.common.aop.domainEvent.DomainEvent

data class EventCreationEvent(
    val hostId: Long?,
    val eventName: String?,
) : DomainEvent() {
    companion object {
        @JvmStatic
        fun of(hostId: Long?, eventName: String?): EventCreationEvent =
            EventCreationEvent(hostId = hostId, eventName = eventName)
    }
}
